EIPS/eip-6956.md

```diff
@@ -0,0 +1,665 @@
+| Scarce digital twin | ASSET | ANY | incompatible | required | - |
+| Lendable digital twin         | OWNER_AND_ASSET | ASSET | required | - | - |
+| **Managing Ownership** |
+| Securing ownership from theft   | OWNER or OWNER_AND_ASSET | ANY | optional | - | required |
+| Selling an house with a mortage  | ASSET  or OWNER_AND_ASSET | ANY | optional | optional | required |
+| Selling a house with a lease | ASSET or OWNER_AND_ASSET | ANY | optional | optional | required |
+| Buying a brand new car with downpayment | ASSET or OWNER_AND_ASSET | ANY | optional | optional | required |
+| Buying a barrel of oil by forward transaction | ASSET or OWNER_AND_ASSET | ANY | optional | optional | required |
+
+
+Legend:
+
+- required ... we don't see a way how to implement the use-case without it
+- incompatible ... this MUSTN'T be implemented, as it is a security risk for the use-case
+- optional ... this MAY optionally be implemented
+
+
+## Backwards Compatibility
+
+---- NEEDS FURTHER DISCUSSION ----
+
+No backward compatibility issues found. 
+
+This EIP is fully compatible with ERC-721 and (when extended with the `IERC6956Floatable`-interface) corresponds to the well-known ERC-721 behavior with an additional authorization-mechanism via attestations. Therefore we recommend - especially for physical assets - to use the present EIP instead of ERC-721 and amend it with extensions designed for ERC-721.
+
+However, it is RECOMMENDED to extend implementations of the proposed standard with an interface indicating transferability of NFTs for market places. Examples include [ERC-6454](eip-6454.md) and [ERC-5484](eip-5484.md).
+
+Many ERC-721 extensions suggest to add additional throw-conditions to transfer methods. This standard is fully compatible, as
+
+- The often-used ERC-721 `_beforeTokenTransfer()` hook must be called for all transfers including attestation-authorized transfers.
+- A `_beforeAnchorUse()` hook is suggested in the reference implementation, which only is called when using attestation as authorization.
+
+
+
+## Test Cases
+
+Test cases are available:
+
+- For only implementing [the proposed standard interface](../assets/eip-6956/contracts/IERC6956.sol) can be found [here](../assets/eip-6956/test/ERC6956.ts)
+- For implementing [the proposed standard interface](../assets/eip-6956/contracts/IERC6956.sol), [the Floatable extension](../assets/eip-6956/contracts/IERC6956Floatable.sol) and [the AttestationLimited extension](../assets/eip-6956/contracts/IERC6956AttestationLimited.sol) can be found [here](../assets/eip-6956/test/ERC6956Full.ts)
+
+## Reference Implementation
+
+The reference implementations are [MIT](../assets/eip-6956/LICENSE.md) licensed. They can be freely used with digital assets as defined in the present EIP. No general license is granted to the *commercial* use or application of the reference implementation or derived works with physical assets as defined in the present EIP. Visit metaanchor.io for further details. 
```

Comment:
Normally we don't note licensing information in the EIP itself. You're free to include the MIT license in your assets directory though.

---

I am not a lawyer, but if the reference implementation is available under the MIT license, then it _is_ licensed for commercial use as long as you comply with the terms of the MIT license, no?

---

While [we're still working out the specifics](https://ethereum-magicians.org/t/licences-for-assets-reference-implementations-in-eips/14329), we only allow [OSI Approved Licenses](https://opensource.org/licenses/). MIT with this non-commercial requirement is not an approved license.

---

Please remove the link to `metaanchor.io`.

Reply:
No problem for me to remove it (will do!). 

However and adding to the [forum discussion](https://ethereum-magicians.org/t/licences-for-assets-reference-implementations-in-eips/14329) (will copy it there as well)

To include that limitation was recommendation from our lawyer though. Not to protect us as the authors but mainly to protect users of the reference implementation. Let me quickly outline our intentions;

- For disclosure, we (Authentic Vision, the company I'm working for) offer with our MetaAnchor-Technology (featuring a random hologram and smartphone app, can be seen [in 10sec here](https://youtu.be/bz-sgusrcVA?t=52) ) a setup that is suitable to act as anchor-technology and oracle. We also own patents for that technology.
- Still we want everyone (commercially and non-commercially) to be able to use the reference implementation without limits.
- The rationale of the lawyer was, that many anchor-technologies (such as our MetaAnchor, NFCs, special QR-Codes, ...) will likely be patent-protected. He recommended to add (especially for commercial) a clause to remind people to be mindful when using the reference implementation in combination with physical assets and physical anchor technologies. Just because they're using the reference implementation does not grant them the right to use any arbitrary anchor technology. 
- The Reference Implementation is MIT, because some of our patents may have similarities to parts of the EIP in some embodiments - we want to be very clear that people can use the EIP and reference implementation freely and unlimited, even though parts may be mentioned in our patents.

==> I would argue that if we were to enforce CC0 for reference implementations without amendments, I would have no legal framework to ensure people it is safe to use our reference implementation here. That's why the reference implementation must be MIT or similar imo.

Reply:
Removed in https://github.com/ethereum/EIPs/pull/6956/commits/1713073d9c204829c446afffa7afd4fbd7754bd6

Reply:
Neither is possible unfortunately
- I cannot list patents I'm aware of, as I would - depending on which legislation we look at - be liable for the completeness of this list. Also, it would be legal advice, which I cannot give. 
- Patents are typically pending for years, including many of ours. So even if I wanted to, I cannot grant licenses on the grounds of a not-yet granted patent. 

With this whole thing we wanted to reassure people and at the same time raise awareness, that they could be infringing other patents (not ours) especially on the physical side.

But I think maybe we may be seeing things too complicated or hypothetical here. Simply licensing the reference implementation under MIT should do the job and be the most reassuring thing we can do. What do you think?

Reply:
Following discussion in Discord, should be resolved imho via https://github.com/ethereum/EIPs/commit/830195450c3883a1f9541840d1f1b8ffeb5ae140